Solution1/DataAccessLayer/Repository/WeatherRepository.cs

```diff
@@ -0,0 +1,28 @@
+﻿using DataAccessLayer.Configuration;
+using DataAccessLayer.Models;
+using DataAccessLayer.Repository.Abstract;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Repository
+{
+    public class WeatherRepository : IWeatherRepository
+    {
+        readonly IDbContextFactory<RepositoryContext> _repositoryFactory;
+
+        public WeatherRepository(IDbContextFactory<RepositoryContext> repositoryFactory)
+        {
+            _repositoryFactory = repositoryFactory;
+        }
+
+        public async Task AddWeatherAsync (Weather weather)
+        {
+            using (var repository = _repositoryFactory.CreateDbContext())
```

Comment:
so your WeatherRepository is responsible for creating context, where is single responsibility? you don't need IDbContextFactory, context should be registered in Startup with special method

Reply:
+